### ISSUE TITLE
Dropdowns in Navbar

### ISSUE THREAD

**Issue by hwaterke** [reactstrap/reactstrap#326]
I am not sure how to include a dropdown in a navbar when using the Navbar component.
Is it supported? Could you provide an example?

Thanks a lot in advance.

**Comment by TheSharpieOne** [comment]
See #304

**Comment by eddywashere** [comment]
This comes up often enough to where it should probably get added to docs as an additional navbar example. Hoping to cover the typical use cases. 

- nav without dropdown, without responsive collapsing
- nav without dropdown, with responsive collapsing
- nav with dropdown, with responsive collapsing

**Comment by TheSharpieOne** [comment]
I would also like to point out some of the inconsistencies around this. For instance, some components will accept a `nav` prop which [adds the extra class](https://github.com/reactstrap/reactstrap/blob/master/src/DropdownToggle.js#L64), while others have a completely separate component which [wraps the 'base' component](https://github.com/reactstrap/reactstrap/blob/master/src/NavDropdown.js#L31) to [add the extra class](https://github.com/reactstrap/reactstrap/blob/master/src/NavDropdown.js#L27)
I am a fan of using the prop to trigger the class, it's much easier to use (no need to change imports) as well as document since it isn't a different component.

tl;dr: `NavDropdown` can be killed in favor of `Dropdown` with `nav` prop.

P.S. Along the same lines, maybe `ButtonDropdown` can be killed in favor a `button` prop?

**Comment by eddywashere** [comment]
Thanks for the reminder : ]

+1 to moving to props vs wrapper components. Docs should get updated to use the prop method now and maybe we should add deprecation note on those wrapper components before removing them.

**Comment by supergibbs** [comment]
Releated, I don't think nav dropdowns work correctly. In the bootstrap docs, they stay within the navbar when collapsed.
